packages/utils/src/internals/extract-microdata.ts

```diff
@@ -0,0 +1,72 @@
+import type { Dictionary } from '@crawlee/types';
+import type { CheerioAPI } from 'cheerio';
+import { load } from 'cheerio';
+
+/**
+ * Extract schema.org microdata from an HTML document using Cheerio.
+ *
+ * @param $ A `CheerioAPI` instance OR raw HTML string.
+ * @returns Extracted metadata as a Dictionary.
+ */
+export function extractMicrodata(raw: string): Dictionary<any>;
+export function extractMicrodata($: CheerioAPI): Dictionary<any>;
+export function extractMicrodata(_item: CheerioAPI | string): Dictionary<any> {
+    const $ = typeof _item === 'string' ? load(_item) : _item;
+
+    const extractValue = (elem: any) => {
+        return $(elem).attr('content') || $(elem).text()?.trim() || $(elem).attr('src') || $(elem).attr('href') || null;
+    };
```

Comment:
This is not entirely correct as per the [microdata specification](https://html.spec.whatwg.org/multipage/microdata.html). See point [5.2.4 Values](https://html.spec.whatwg.org/multipage/microdata.html#values) - the type of the value extracted is based on the element type.